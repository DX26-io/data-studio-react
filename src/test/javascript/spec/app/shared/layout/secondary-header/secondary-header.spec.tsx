import React from 'react';
import { shallow } from 'enzyme';

import { Breadcrumbs, Item, Button } from '@adobe/react-spectrum';

import SecondaryHeader, { ISecondaryHeaderProps } from 'app/shared/layout/secondary-header/secondary-header';

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

describe('SecondaryHeader', () => {
  let mountedWrapper;

  const defaultProps: ISecondaryHeaderProps = {
    breadcrumbItems: [
      { label: 'Home', route: '/' },
      { label: 'Dashboards', route: '/dashboards' },
      { label: 'Inventory Dashboard', route: '/dashboards/d123' },
      { label: 'Stock', route: '/dashboards/d123/v123' },
    ],
    title: 'SH Unit Test',
  };

  const wrapper = (props = defaultProps) => {
    if (!mountedWrapper) {
      mountedWrapper = shallow(<SecondaryHeader {...props} />);
    }
    return mountedWrapper;
  };

  const wrapperWithChildren = (props = defaultProps) => {
    if (!mountedWrapper) {
      mountedWrapper = shallow(
        <SecondaryHeader {...props}>
          <Button variant="primary" marginX="size-150">
            Edit
          </Button>
          <Button variant="secondary">Save</Button>
        </SecondaryHeader>
      );
    }
    return mountedWrapper;
  };

  beforeEach(() => {
    mountedWrapper = undefined;
  });

  it('Renders Secondary Header with default Props', () => {
    const component = wrapper();
    expect(component).toMatchSnapshot();
  });

  it('Renders Secondary Header with children', () => {
    const component = wrapperWithChildren();
    expect(component).toMatchSnapshot();
  });

  it('Validate Breadcrumb Items', () => {
    const breadcrumbs = wrapper().find(Breadcrumbs);
    const breadcrumbItems = breadcrumbs.find(Item);
    expect(breadcrumbItems.length).toEqual(defaultProps.breadcrumbItems.length);
  });
});
