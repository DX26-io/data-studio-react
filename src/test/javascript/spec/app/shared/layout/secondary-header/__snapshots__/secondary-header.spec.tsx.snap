// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryHeader Renders Secondary Header with children 1`] = `
<ForwardRef($cb4e2cb36ae5e3ab01f62ab7d1672c67$var$View)
  backgroundColor="default"
  paddingX="size-150"
  paddingY="size-100"
>
  <ForwardRef($f784fbb881b5b101cd7607722a86$var$Grid)
    areas={
      Array [
        "breadcrumbs title options",
      ]
    }
    columns={
      Array [
        "1fr",
        "1fr",
        "1fr",
      ]
    }
    maxHeight="size-600"
  >
    <ForwardRef($e9098d18abf2a118bc3bc95f0d402bb5$var$Flex)
      alignItems="center"
      gridArea="breadcrumbs"
      justifyContent="start"
    >
      <ForwardRef($e2f01fb373fadb4c4bc7c0c3ead285a$var$Breadcrumbs)
        size="M"
      >
        <$de85adf0b38f4f13e9cffd2f21$var$Item
          key="home"
        >
          <Link
            to="/"
          >
            Home
          </Link>
        </$de85adf0b38f4f13e9cffd2f21$var$Item>
        <$de85adf0b38f4f13e9cffd2f21$var$Item
          key="dashboards"
        >
          <Link
            to="/dashboards"
          >
            Dashboards
          </Link>
        </$de85adf0b38f4f13e9cffd2f21$var$Item>
        <$de85adf0b38f4f13e9cffd2f21$var$Item
          key="d123"
        >
          <Link
            to="/dashboards/d123"
          >
            Inventory Dashboard
          </Link>
        </$de85adf0b38f4f13e9cffd2f21$var$Item>
        <$de85adf0b38f4f13e9cffd2f21$var$Item
          key="v123"
        >
          <Link
            to="/dashboards/d123/v123"
          >
            Stock
          </Link>
        </$de85adf0b38f4f13e9cffd2f21$var$Item>
      </ForwardRef($e2f01fb373fadb4c4bc7c0c3ead285a$var$Breadcrumbs)>
    </ForwardRef($e9098d18abf2a118bc3bc95f0d402bb5$var$Flex)>
    <ForwardRef($e9098d18abf2a118bc3bc95f0d402bb5$var$Flex)
      alignItems="center"
      gridArea="title"
      justifyContent="center"
    >
      <ForwardRef($a76c0421a0242146b7d397a8ac50$var$Heading)
        level={3}
        marginY="size-0"
      >
        SH Unit Test
      </ForwardRef($a76c0421a0242146b7d397a8ac50$var$Heading)>
    </ForwardRef($e9098d18abf2a118bc3bc95f0d402bb5$var$Flex)>
    <ForwardRef($e9098d18abf2a118bc3bc95f0d402bb5$var$Flex)
      alignItems="center"
      gridArea="options"
      justifyContent="end"
    >
      <ForwardRef($ead2ca032915257adf53f639d99d1a73$var$Button)
        marginX="size-150"
        variant="primary"
      >
        Edit
      </ForwardRef($ead2ca032915257adf53f639d99d1a73$var$Button)>
      <ForwardRef($ead2ca032915257adf53f639d99d1a73$var$Button)
        variant="secondary"
      >
        Save
      </ForwardRef($ead2ca032915257adf53f639d99d1a73$var$Button)>
    </ForwardRef($e9098d18abf2a118bc3bc95f0d402bb5$var$Flex)>
  </ForwardRef($f784fbb881b5b101cd7607722a86$var$Grid)>
</ForwardRef($cb4e2cb36ae5e3ab01f62ab7d1672c67$var$View)>
`;

exports[`SecondaryHeader Renders Secondary Header with default Props 1`] = `
<ForwardRef($cb4e2cb36ae5e3ab01f62ab7d1672c67$var$View)
  backgroundColor="default"
  paddingX="size-150"
  paddingY="size-100"
>
  <ForwardRef($f784fbb881b5b101cd7607722a86$var$Grid)
    areas={
      Array [
        "breadcrumbs title options",
      ]
    }
    columns={
      Array [
        "1fr",
        "1fr",
        "1fr",
      ]
    }
    maxHeight="size-600"
  >
    <ForwardRef($e9098d18abf2a118bc3bc95f0d402bb5$var$Flex)
      alignItems="center"
      gridArea="breadcrumbs"
      justifyContent="start"
    >
      <ForwardRef($e2f01fb373fadb4c4bc7c0c3ead285a$var$Breadcrumbs)
        size="M"
      >
        <$de85adf0b38f4f13e9cffd2f21$var$Item
          key="home"
        >
          <Link
            to="/"
          >
            Home
          </Link>
        </$de85adf0b38f4f13e9cffd2f21$var$Item>
        <$de85adf0b38f4f13e9cffd2f21$var$Item
          key="dashboards"
        >
          <Link
            to="/dashboards"
          >
            Dashboards
          </Link>
        </$de85adf0b38f4f13e9cffd2f21$var$Item>
        <$de85adf0b38f4f13e9cffd2f21$var$Item
          key="d123"
        >
          <Link
            to="/dashboards/d123"
          >
            Inventory Dashboard
          </Link>
        </$de85adf0b38f4f13e9cffd2f21$var$Item>
        <$de85adf0b38f4f13e9cffd2f21$var$Item
          key="v123"
        >
          <Link
            to="/dashboards/d123/v123"
          >
            Stock
          </Link>
        </$de85adf0b38f4f13e9cffd2f21$var$Item>
      </ForwardRef($e2f01fb373fadb4c4bc7c0c3ead285a$var$Breadcrumbs)>
    </ForwardRef($e9098d18abf2a118bc3bc95f0d402bb5$var$Flex)>
    <ForwardRef($e9098d18abf2a118bc3bc95f0d402bb5$var$Flex)
      alignItems="center"
      gridArea="title"
      justifyContent="center"
    >
      <ForwardRef($a76c0421a0242146b7d397a8ac50$var$Heading)
        level={3}
        marginY="size-0"
      >
        SH Unit Test
      </ForwardRef($a76c0421a0242146b7d397a8ac50$var$Heading)>
    </ForwardRef($e9098d18abf2a118bc3bc95f0d402bb5$var$Flex)>
    <ForwardRef($e9098d18abf2a118bc3bc95f0d402bb5$var$Flex)
      alignItems="center"
      gridArea="options"
      justifyContent="end"
    />
  </ForwardRef($f784fbb881b5b101cd7607722a86$var$Grid)>
</ForwardRef($cb4e2cb36ae5e3ab01f62ab7d1672c67$var$View)>
`;
